File: Selenium/Project/ProjectActivity7.java
package Selenium_Java_Project;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.WebElement;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.support.locators.RelativeLocator;
import org.openqa.selenium.support.ui.ExpectedConditions;
import org.openqa.selenium.support.ui.Select;
import org.openqa.selenium.support.ui.WebDriverWait;
import org.testng.Assert;

import java.time.Duration;

public class ProjectActivity7 {

    public static void main(String args[]) {

        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        WebDriver driver = new FirefoxDriver();
        WebDriverWait wait = new WebDriverWait(driver, Duration.ofSeconds(10));

        driver.get("https://alchemy.hguy.co/jobs/");

        driver.findElement(By.xpath("//a[text()='Post a Job']")).click();

        driver.findElement(By.id("create_account_email")).sendKeys("rajshekarboyini@y.com");

        driver.findElement(By.id("job_title")).sendKeys("Test specialist");

        driver.findElement(By.id("job_location")).sendKeys("UK");

        WebElement jobTypeDropDown = driver.findElement(By.id("job_type"));
        jobTypeDropDown.click();

        Select selectJobType = new Select(jobTypeDropDown);
        selectJobType.selectByVisibleText("Full Time");

        //wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//body[contains(@data-id, 'job_description')]")));

        driver.findElement(By.xpath("//iframe[@id= 'job_description_ifr']")).sendKeys("Automation Testor");

        driver.findElement(By.xpath("//input[@id='application']")).sendKeys("https://alchemy.hguy.co/jobs/");

        driver.findElement(By.id("company_name")).sendKeys("IBM");

        driver.findElement(By.name("submit_job")).click();

        //wait.until(ExpectedConditions.visibilityOfElementLocated(By.xpath("//input[@id='job_preview_submit_button']")));

        driver.findElement(By.xpath("//input[@id='job_preview_submit_button']")).click();

        driver.findElement(By.xpath("//a[text()='click here']")).click();

        String JobTitleText =  driver.findElement(By.xpath("//h1[contains(@class, 'entry-title')]")).getText();

        Assert.assertEquals(JobTitleText, "Test specialist");

        driver.quit();





    }
}

File: Python/Activities/ProjectActivity7.py
from selenium import webdriver
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService
from selenium.webdriver.support.select import Select
from selenium.webdriver.support.wait import WebDriverWait
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service1 = FirefoxService(GeckoDriverManager().install())


# Start the Driver
with webdriver.Firefox(service=service1) as driver:

    wait = WebDriverWait(driver, 10)
    # Open the browser to the URL
    driver.get("https://alchemy.hguy.co/jobs")
    title = driver.title

    print("Page title is : ", driver.title)

    driver.find_element(By.XPATH, "//a[text()='Post a Job']").click()

    driver.find_element(By.XPATH, "//input[@id='create_account_email']").send_keys("raj@gmail.com")

    driver.find_element(By.XPATH, "//input[@id='job_title']").send_keys("Test specialist")

    driver.find_element(By.XPATH, "//input[@id='job_location']").send_keys("UK")

    jobType = driver.find_element(By.ID, "job_type")
    dropdown = Select(jobType)

    #wait.until()

    dropdown.select_by_value("Full Time")

    driver.find_element(By.XPATH, "//iframe[@id='job_description_ifr']").send_keys("Automating the app")

    driver.find_element(By.XPATH, "//input[@id='application']").send_keys("raj@gmail.com")

    driver.find_element(By.XPATH, "//input[@id='company_name']").send_keys("Google")

    driver.find_element(By.XPATH, "//input[@name='submit_job']").click()

    driver.find_element(By.XPATH, "//input[@id='job_preview_submit_button']").click()

    driver.find_element(By.XPATH, "//a[text()='click here']").click()
    job = driver.find_element(By.XPATH, "//h1[contains(@class, 'entry-title')]")
    jobTitle = job.text

    assert jobTitle == "Test specialist"


